openfl/utilities/dataloading.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2020-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from typing import Union
+
+from openfl.federated import Plan
+from openfl.utilities.mocks import MockDataLoader
+from openfl.federated.data.loader import DataLoader
+
+def get_dataloader(
+    plan: Plan,
+    prefer_minimal: bool = False,
+    input_shape: Union[list, dict] = None
+) -> DataLoader:
+    """Get dataloader instance from plan
+
+    Args:
+        plan (Plan):
+            plan object linked with the dataloader
+        prefer_minimal (bool ?):
+            prefer to use MockDataLoader which can be used to more easily
+            instantiate task_runner without any initial data
+        input_shape (list | dict ?):
+            input_shape denoted by list notation `[a,b,c, ...]` or in case
+            of multihead models, dict object with individual layer keys such
+            as `{"input_0": [a,b,...], "output_1": [x,y,z, ...]}`
+    """
+    data_loader = None
+    
+    # try bypassing initial data requirement if input_shape is knowable
+    if input_shape is None and \
+        plan.config['data_loader'] and \
+       'input_shape' in plan.config['data_loader']:
+        input_shape = plan.config['data_loader']['input_shape']
+
+    if prefer_minimal and input_shape:
+        # input_shape known, we can use MockDataLoader
+        data_loader: DataLoader = MockDataLoader(input_shape)
+    else:
+        # input_shape not known, load from data of first collab
+        # note: this is very odd flow
+        # TODO: need to review this way of getting data_loader
```

Comment:
one edge case if when user provides data config and doesn't provide collaborator yaml since the PR doesn't mandate it can break this logic

Reply:
the rest of the updates looks good. Can you please review this edge case

Reply:
Updated and handled both `data.yaml` not present case and also actual data path given by the selected collaborator not existing in the current workspace.